lib/healthcheck/api_tokens.rb

```diff
@@ -0,0 +1,47 @@
+module Healthcheck
+  class ApiTokens
+    WARNING_THRESHOLD = 1.year.to_i
+    CRITICAL_THRESHOLD = 1.month.to_i
+
+    QUERY = <<-SQL.freeze
+      SELECT users.name, oauth_applications.name FROM oauth_access_tokens
+      INNER JOIN users ON users.id = resource_owner_id
+      INNER JOIN oauth_applications ON oauth_applications.id = application_id
+      WHERE oauth_access_tokens.revoked_at IS NULL
+      AND users.api_user = TRUE
+      AND (oauth_access_tokens.expires_in -
+          (unix_timestamp(now()) -
+           unix_timestamp(oauth_access_tokens.created_at))) < %<threshold>s
+    SQL
+
+    def name
+      :api_tokens
+    end
+
+    def details
+      { warnings: warnings - criticals, criticals: criticals }
```

Comment:
It'd be nice if these were a bit more readable and said when the expiry was. E.g. "#{user_name} token for #{publishing_app} expires at #{datetime}" 

I don't think we'd actually need to break it up into warning and critical groups if we have the times there - what do you think?

Reply:
Done :-). Rewritten a bit to avoid calculating the time twice. I also thought it would be clearer to show the number of days, instead of the timestamp.